### ISSUE TITLE
How to use customized template?

### ISSUE THREAD

**Issue by mkjeff** [NSwag/NSwag#44]
built-in angular2 typescript template is not suitable for me. how to use customized template?

**Comment by rsuter** [comment]
This is currently not possible, because the interface between the model and template is not very stable.. What exactly is missing/wrong in the angular2 template? Can the existing template be enhanced to support your needs?

**Comment by cwoolum** [comment]
I would be interested in a custom template as well. My specific example is that I need to inject a configuration service into my class to get the base url rather than force the Schema url to be the base url.

**Comment by mkjeff** [comment]
Client constructor parameter: baseUrl should be injected.
use OpaqueToken:

```ts
const API_BASE_URL: OpaqueToken = new OpaqueToken('API_BASE_URL');

constructor(
    /* no need @Inject(Http)*/ 
    http: Http, 
    @Inject(API_BASE_URL)@Optional() baseUrl?: string){
  ...
}
```

**Comment by rsuter** [comment]
Injectable base url added in https://github.com/NSwag/NSwag/commit/86ba0d1c7074605359f449136d2b5cd8d86995bf

I left the inject attributes for clarity. Ill open a new issue for customized templates because it is a bigger task (ie portable tpl engine, tpl fallbacks and stable models)

**Comment by rsuter** [comment]
And https://github.com/NSwag/NSwag/commit/76795834d9cc12785a04074b0c4b44d22853b83e

**Comment by rsuter** [comment]
Added new issue: https://github.com/NSwag/NSwag/issues/51

**Comment by rsuter** [comment]
Available in v2.1

**Comment by rsuter** [comment]
Missed the imports and export keyword, check v2.2
